File: platform-console/apps/raap-admin/src/api/core/prompt-debug.ts
import { requestClient } from '#/api/request';

export namespace PromptDebugApi {
  export interface RunRequest {
    prompt: string;
    model_code: string;
    temperature?: number;
    max_tokens?: number;
    thinking_mode?: 'default' | 'disabled' | 'enabled';
    system_prompt?: string;
    run_group_id?: string;
    workbench_mode?: 'compare' | 'single';
    panel_key?: 'left' | 'right';
    item_index?: number;
    batch_size?: number;
  }

  export interface TokenUsage {
    input_tokens: number;
    output_tokens: number;
    total_tokens: number;
  }

  export interface RunResponse {
    success: boolean;
    content?: null | string;
    model_code?: null | string;
    provider_code?: null | string;
    provider_model?: null | string;
    usage?: null | TokenUsage;
    latency_ms?: null | number;
    error_message?: null | string;
    history_id?: null | number;
    run_group_id?: null | string;
  }

  export interface HistoryItem {
    id: number;
    run_group_id: string;
    workbench_mode: 'compare' | 'single';
    panel_key: 'left' | 'right';
    item_index: number;
    batch_size: number;
    prompt: string;
    system_prompt?: null | string;
    requested_model_code: string;
    temperature: number;
    max_tokens: number;
    thinking_mode: 'default' | 'disabled' | 'enabled';
    success: boolean;
    content?: null | string;
    model_code?: null | string;
    provider_code?: null | string;
    provider_model?: null | string;
    token_usage?: null | TokenUsage;
    latency_ms?: null | number;
    error_message?: null | string;
    create_time?: null | string;
  }

  export interface HistoryGroupSummary {
    run_group_id: string;
    workbench_mode: 'compare' | 'single';
    create_time?: null | string;
    total_count: number;
    success_count: number;
    failed_count: number;
    panel_keys: Array<'left' | 'right'>;
    model_codes: string[];
    prompt_preview: string;
  }

  export interface HistoryListResponse {
    items: HistoryGroupSummary[];
  }

  export interface HistoryGroupDetail {
    run_group_id: string;
    workbench_mode: 'compare' | 'single';
    create_time?: null | string;
    records: HistoryItem[];
  }
}

export async function runPromptDebugApi(data: PromptDebugApi.RunRequest) {
  return requestClient.post<PromptDebugApi.RunResponse>(
    '/v1/content-agent/prompt-debug/run',
    data,
    {
      timeout: 180_000,
    },
  );
}

export async function getPromptDebugHistoryApi(limit = 30) {
  return requestClient.get<PromptDebugApi.HistoryListResponse>(
    '/v1/content-agent/prompt-debug/history',
    { params: { limit } },
  );
}

export async function getPromptDebugHistoryDetailApi(runGroupId: string) {
  return requestClient.get<PromptDebugApi.HistoryGroupDetail>(
    `/v1/content-agent/prompt-debug/history/${runGroupId}`,
  );
}

File: platform-server/app/schemas/prompt_debug.py
"""Schemas for the MAGA prompt debug workbench."""
from datetime import datetime
from typing import Optional
from typing import Literal

from pydantic import Field, field_validator

from app.schemas.base import BaseSchema


class PromptDebugRequest(BaseSchema):
    """One raw prompt invocation from the operator debug workbench."""

    prompt: str = Field(..., min_length=1, description="用户提示词")
    model_code: str = Field(..., min_length=1, description="统一模型编码")
    temperature: Optional[float] = Field(default=None, ge=0, le=2)
    max_tokens: Optional[int] = Field(default=None, ge=1, le=20000)
    thinking_mode: Literal["default", "enabled", "disabled"] = Field(
        default="default",
        description="模型思考模式；default 不发送 thinking 参数",
    )
    system_prompt: Optional[str] = Field(default=None)
    run_group_id: Optional[str] = Field(default=None, min_length=1, max_length=64)
    workbench_mode: Literal["single", "compare"] = "single"
    panel_key: Literal["left", "right"] = "left"
    item_index: int = Field(default=0, ge=0, le=19)
    batch_size: int = Field(default=1, ge=1, le=20)

    @field_validator("prompt", "model_code")
    @classmethod
    def _strip_required_text(cls, value: str) -> str:
        text = value.strip()
        if not text:
            raise ValueError("不能为空")
        return text

    @field_validator("system_prompt")
    @classmethod
    def _strip_optional_text(cls, value: Optional[str]) -> Optional[str]:
        if value is None:
            return None
        text = value.strip()
        return text or None


class PromptDebugTokenUsage(BaseSchema):
    input_tokens: int = 0
    output_tokens: int = 0
    total_tokens: int = 0


class PromptDebugResponse(BaseSchema):
    success: bool
    content: Optional[str] = None
    model_code: Optional[str] = None
    provider_code: Optional[str] = None
    provider_model: Optional[str] = None
    usage: Optional[PromptDebugTokenUsage] = None
    latency_ms: Optional[int] = None
    error_message: Optional[str] = None
    history_id: Optional[int] = None
    run_group_id: Optional[str] = None


class PromptDebugHistoryItem(BaseSchema):
    id: int
    run_group_id: str
    workbench_mode: Literal["single", "compare"]
    panel_key: Literal["left", "right"]
    item_index: int
    batch_size: int
    prompt: str
    system_prompt: Optional[str] = None
    requested_model_code: str
    temperature: float
    max_tokens: int
    thinking_mode: Literal["default", "enabled", "disabled"] = "default"
    success: bool
    content: Optional[str] = None
    model_code: Optional[str] = None
    provider_code: Optional[str] = None
    provider_model: Optional[str] = None
    token_usage: Optional[dict] = None
    latency_ms: Optional[int] = None
    error_message: Optional[str] = None
    create_time: Optional[datetime] = None


class PromptDebugHistoryGroupSummary(BaseSchema):
    run_group_id: str
    workbench_mode: Literal["single", "compare"]
    create_time: Optional[datetime] = None
    total_count: int
    success_count: int
    failed_count: int
    panel_keys: list[Literal["left", "right"]] = Field(default_factory=list)
    model_codes: list[str] = Field(default_factory=list)
    prompt_preview: str = ""


class PromptDebugHistoryListResponse(BaseSchema):
    items: list[PromptDebugHistoryGroupSummary] = Field(default_factory=list)


class PromptDebugHistoryGroupDetail(BaseSchema):
    run_group_id: str
    workbench_mode: Literal["single", "compare"]
    create_time: Optional[datetime] = None
    records: list[PromptDebugHistoryItem] = Field(default_factory=list)
